contracts/src/plugin/extensions/proposal/IProposal.sol

```diff
@@ -23,15 +24,51 @@ interface IProposal {
         uint64 startDate,
         uint64 endDate,
         bytes metadata,
-        IDAO.Action[] actions,
+        Action[] actions,
         uint256 allowFailureMap
     );
 
     /// @notice Emitted when a proposal is executed.
     /// @param proposalId The ID of the proposal.
     event ProposalExecuted(uint256 indexed proposalId);
 
+    /// @notice Creates a new proposal.
+    /// @param metadata The metadata of the proposal.
+    /// @param actions The actions that will be executed after the proposal passes.
+    /// @param startDate The start date of the proposal.
+    /// @param endDate The end date of the proposal.
+    /// @param data The additional abi-encoded data to include more necessary fields.
+    /// @return proposalId The id of the proposal.
+    function createProposal(
+        bytes memory metadata,
+        Action[] memory actions,
+        uint64 startDate,
+        uint64 endDate,
+        bytes memory data
+    ) external returns (uint256 proposalId);
+
+    /// @notice Whether proposal can be executed or not.
+    /// @param proposalId The id of the proposal.
+    /// @return bool Returns if proposal can be executed or not.
+    function canExecute(uint256 proposalId) external view returns (bool);
+
+    /// @notice Creates a proposal Id.
+    /// @param actions The actions that will be executed after the proposal passes.
+    /// @param metadata The custom metadata that is passed when creating a proposal.
+    /// @return proposalId The id of the proposal.
+    function createProposalId(
+        Action[] memory actions,
+        bytes memory metadata
```

Comment:
Why are these parameters needed here?
- If it is to create an ID using them as entropy, why aren't we also receiving `data`?